### ISSUE TITLE
Consumer concurrency not working (all consumed message by same thread)

### ISSUE THREAD

**Issue by lubos-bistak** [spring-cloud/spring-cloud-stream-binder-kafka#356]
Hello,

i built an application that writes and reads Java objects (serialized/deserialized using Avro) into a Kafka topic having 3 partitions by using Spring Cloud Stream. I want to read in parallel from this topic, but it seems not be working as always same thread handles the Java objects from this topic sequential.

Code to read from the topic looks like this:
```
@StreamListener(InputSink.INPUT_TOPIC1)
    public void readFromTopic1(SomeObject obj) {
        log.debug("[] Read from Kafka: {} ", Thread.currentThread().getName(),obj.toString());
        try {
            Thread.sleep(100);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
```
In logs i always see the same thread name "-L-1":
`-L-1 Read from Kafka: ...`

Application start-up logs:
```
[           -C-1] o.s.c.s.b.k.KafkaMessageChannelBinder$3  : partitions assigned:[topic1-2]
[           -C-1] o.s.c.s.b.k.KafkaMessageChannelBinder$3  : partitions assigned:[topic1-0]
[           -C-1] o.s.c.s.b.k.KafkaMessageChannelBinder$3  : partitions assigned:[topic1-1]
```

Configuration in application.yml:
```
spring:
  cloud:
    stream:
      default:
        contentType: application/*+avro
      bindings:
        outputTopic1
          destination: topic1
        inputTopic1:
          destination: topic1
          group: group-inputTopic1
          consumer:
            concurrency: 3
            partitioned: true
      schema:
        avro:
          dynamic-schema-generation-enabled: true
```

Spring dependencies:
* spring-cloud-stream-reactive:1.3.0.RELEASE
* spring-could-stream-schema:1.3.0.RELEASE
* spring-cloud-starter-stream-kafka:1.3.0.RELEASE
* spring-cloud-stream:1.3.0.RELEASE
* spring-boot-starter-web:1.5.10.RELEASE

Thanks a lot for any advice !!!

**Comment by garyrussell** [comment]
This is just a thread naming problem; 1.3.x inherits a very old version of spring-kafka from boot. The messages are actually processed on different threads...

```java
@SpringBootApplication
@EnableBinding(MyChannels.class)
public class Kbgh356Application {

	private final Logger logger = LoggerFactory.getLogger(Kbgh356Application.class);

	public static void main(String[] args) {
		SpringApplication.run(Kbgh356Application.class, args);
	}

	@Bean
	public ApplicationRunner runner(MessageChannel output) {
		return args -> {
			Thread.sleep(10_000);
			output.send(new GenericMessage<>("foo", Collections.singletonMap(KafkaHeaders.PARTITION_ID, 0)));
			output.send(new GenericMessage<>("bar", Collections.singletonMap(KafkaHeaders.PARTITION_ID, 1)));
			output.send(new GenericMessage<>("baz", Collections.singletonMap(KafkaHeaders.PARTITION_ID, 2)));
		};
	}

	@StreamListener("input")
	public void listen(String in, @Header(KafkaHeaders.RECEIVED_PARTITION_ID) int partition) {
		logger.info(in + " from partition " + partition + " thread: " + Thread.currentThread().getId());
	}

}

interface MyChannels {

	@Input
	MessageChannel input();

	@Output
	MessageChannel output();

}
```

```
2018-04-16 11:15:55.299  INFO 74942 --- [           -C-1] o.s.c.s.b.k.KafkaMessageChannelBinder$3  : partitions assigned:[topic1-0]
2018-04-16 11:15:55.299  INFO 74942 --- [           -C-1] o.s.c.s.b.k.KafkaMessageChannelBinder$3  : partitions assigned:[topic1-2]
2018-04-16 11:15:55.299  INFO 74942 --- [           -C-1] o.s.c.s.b.k.KafkaMessageChannelBinder$3  : partitions assigned:[topic1-1]
...
2018-04-16 11:16:02.451  INFO 74942 --- [           -L-1] com.example.Kbgh356Application           : bar from partition 1 thread: 61
2018-04-16 11:16:02.451  INFO 74942 --- [           -L-1] com.example.Kbgh356Application           : baz from partition 2 thread: 60
2018-04-16 11:16:02.451  INFO 74942 --- [           -L-1] com.example.Kbgh356Application           : foo from partition 0 thread: 59
```

Notice that although the threads have the same name, they have different IDs.

In any case, I would recommend upgrading to a newer release of the kafka libraries:

```xml
		<dependency>
			<groupId>org.springframework.kafka</groupId>
			<artifactId>spring-kafka</artifactId>
			<version>1.3.4.RELEASE</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.integration</groupId>
			<artifactId>spring-integration-kafka</artifactId>
			<version>2.3.0.RELEASE</version>
		</dependency>
		<dependency>
			<groupId>org.apache.kafka</groupId>
			<artifactId>kafka-clients</artifactId>
			<version>0.11.0.1</version>
		</dependency>
```

This has a better threading model (thanks to KIP-62) - there are no longer any `-L-x` threads, just `-C-x`.

(Also 1.3.2 of the spring-cloud-stream jars from Edgware.SR3).

However, it still suffers from the thread naming problem.

I just made a change to spring-kafka to improve things...

```
2018-04-16 11:30:27.349  INFO 77754 --- [ consumer-0-C-1] com.example.Kbgh356Application           : foo from partition 0 thread: 26
2018-04-16 11:30:27.349  INFO 77754 --- [ consumer-1-C-1] com.example.Kbgh356Application           : bar from partition 1 thread: 28
2018-04-16 11:30:27.349  INFO 77754 --- [ consumer-2-C-1] com.example.Kbgh356Application           : baz from partition 2 thread: 30
```

**Comment by lubos-bistak** [comment]
Thank you for your replay, i updated dependencies as you suggested, but now i get following WARNing and programs stops:

```
o.s.c.s.b.k.p.KafkaTopicProvisioner      : Auto creation of topics is enabled, but Kafka AdminUtils class is not present on the classpath. No topic will be created by the binder
2018-04-19 14:30:04.838  WARN [....,,,] 42482 --- [ad | producer-1] org.apache.kafka.clients.NetworkClient   : Error while fetching metadata with correlation id 1 : {businessPartner=LEADER_NOT_AVAILABLE}
```

**Comment by garyrussell** [comment]
Please provide your complete pom.

**Comment by lubos-bistak** [comment]
Complete dependency runtime tree:

```
$ ./gradlew dependencies --configuration runtime

runtime - Runtime dependencies for source set 'main' (deprecated, use 'runtimeOnly ' instead).
+--- org.springframework.boot:spring-boot-starter-web: -> 1.5.10.RELEASE
|    +--- org.springframework.boot:spring-boot-starter:1.5.10.RELEASE
|    |    +--- org.springframework.boot:spring-boot:1.5.10.RELEASE
|    |    |    +--- org.springframework:spring-core:4.3.14.RELEASE
|    |    |    |    \--- commons-logging:commons-logging:1.2
|    |    |    \--- org.springframework:spring-context:4.3.14.RELEASE
|    |    |         +--- org.springframework:spring-aop:4.3.14.RELEASE
|    |    |         |    +--- org.springframework:spring-beans:4.3.14.RELEASE
|    |    |         |    |    \--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    |         |    \--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    |         +--- org.springframework:spring-beans:4.3.14.RELEASE (*)
|    |    |         +--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    |         \--- org.springframework:spring-expression:4.3.14.RELEASE
|    |    |              \--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:1.5.10.RELEASE
|    |    |    \--- org.springframework.boot:spring-boot:1.5.10.RELEASE (*)
|    |    +--- org.springframework.boot:spring-boot-starter-logging:1.5.10.RELEASE
|    |    |    +--- ch.qos.logback:logback-classic:1.1.11
|    |    |    |    +--- ch.qos.logback:logback-core:1.1.11
|    |    |    |    \--- org.slf4j:slf4j-api:1.7.22 -> 1.7.25
|    |    |    +--- org.slf4j:jcl-over-slf4j:1.7.25
|    |    |    |    \--- org.slf4j:slf4j-api:1.7.25
|    |    |    +--- org.slf4j:jul-to-slf4j:1.7.25
|    |    |    |    \--- org.slf4j:slf4j-api:1.7.25
|    |    |    \--- org.slf4j:log4j-over-slf4j:1.7.25
|    |    |         \--- org.slf4j:slf4j-api:1.7.25
|    |    +--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    \--- org.yaml:snakeyaml:1.17
|    +--- org.springframework.boot:spring-boot-starter-tomcat:1.5.10.RELEASE
|    |    +--- org.apache.tomcat.embed:tomcat-embed-core:8.5.27
|    |    |    \--- org.apache.tomcat:tomcat-annotations-api:8.5.27
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:8.5.27
|    |    \--- org.apache.tomcat.embed:tomcat-embed-websocket:8.5.27
|    |         \--- org.apache.tomcat.embed:tomcat-embed-core:8.5.27 (*)
|    +--- org.hibernate:hibernate-validator:5.3.6.Final
|    |    +--- javax.validation:validation-api:1.1.0.Final
|    |    +--- org.jboss.logging:jboss-logging:3.3.0.Final -> 3.3.1.Final
|    |    \--- com.fasterxml:classmate:1.3.1 -> 1.3.4
|    +--- com.fasterxml.jackson.core:jackson-databind:2.8.10
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.8.0
|    |    \--- com.fasterxml.jackson.core:jackson-core:2.8.10
|    +--- org.springframework:spring-web:4.3.14.RELEASE
|    |    +--- org.springframework:spring-aop:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-beans:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-context:4.3.14.RELEASE (*)
|    |    \--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    \--- org.springframework:spring-webmvc:4.3.14.RELEASE
|         +--- org.springframework:spring-aop:4.3.14.RELEASE (*)
|         +--- org.springframework:spring-beans:4.3.14.RELEASE (*)
|         +--- org.springframework:spring-context:4.3.14.RELEASE (*)
|         +--- org.springframework:spring-core:4.3.14.RELEASE (*)
|         +--- org.springframework:spring-expression:4.3.14.RELEASE (*)
|         \--- org.springframework:spring-web:4.3.14.RELEASE (*)
+--- org.springframework.boot:spring-boot-starter-aop: -> 1.5.10.RELEASE
|    +--- org.springframework.boot:spring-boot-starter:1.5.10.RELEASE (*)
|    +--- org.springframework:spring-aop:4.3.14.RELEASE (*)
|    \--- org.aspectj:aspectjweaver:1.8.13
+--- org.springframework.cloud:spring-cloud-stream: -> 1.3.0.RELEASE
|    +--- org.springframework.boot:spring-boot-starter-actuator:1.5.7.RELEASE -> 1.5.10.RELEASE
|    |    +--- org.springframework.boot:spring-boot-starter:1.5.10.RELEASE (*)
|    |    \--- org.springframework.boot:spring-boot-actuator:1.5.10.RELEASE
|    |         +--- org.springframework.boot:spring-boot:1.5.10.RELEASE (*)
|    |         +--- org.springframework.boot:spring-boot-autoconfigure:1.5.10.RELEASE (*)
|    |         +--- com.fasterxml.jackson.core:jackson-databind:2.8.10 (*)
|    |         +--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |         \--- org.springframework:spring-context:4.3.14.RELEASE (*)
|    +--- org.springframework.boot:spring-boot-starter-web:1.5.7.RELEASE -> 1.5.10.RELEASE (*)
|    +--- org.springframework.boot:spring-boot-starter-validation:1.5.7.RELEASE -> 1.5.10.RELEASE
|    |    +--- org.springframework.boot:spring-boot-starter:1.5.10.RELEASE (*)
|    |    +--- org.apache.tomcat.embed:tomcat-embed-el:8.5.27
|    |    \--- org.hibernate:hibernate-validator:5.3.6.Final (*)
|    +--- org.springframework:spring-messaging:4.3.11.RELEASE -> 4.3.14.RELEASE
|    |    +--- org.springframework:spring-beans:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-context:4.3.14.RELEASE (*)
|    |    \--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    +--- org.springframework.integration:spring-integration-core:4.3.12.RELEASE -> 4.3.14.RELEASE
|    |    +--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-aop:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-context:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-messaging:4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-tx:4.3.14.RELEASE
|    |    |    +--- org.springframework:spring-beans:4.3.14.RELEASE (*)
|    |    |    \--- org.springframework:spring-core:4.3.14.RELEASE (*)
|    |    \--- org.springframework.retry:spring-retry:1.1.3.RELEASE -> 1.2.2.RELEASE
|    |         \--- org.springframework:spring-core:4.3.13.RELEASE -> 4.3.14.RELEASE (*)
|    +--- org.springframework.integration:spring-integration-jmx:4.3.12.RELEASE -> 4.3.14.RELEASE
|    |    \--- org.springframework.integration:spring-integration-core:4.3.14.RELEASE (*)
|    +--- org.springframework:spring-tuple:1.0.0.RELEASE
|    |    +--- com.esotericsoftware:kryo-shaded:3.0.3
|    |    |    +--- com.esotericsoftware:minlog:1.3.0
|    |    |    \--- org.objenesis:objenesis:2.1
|    |    +--- org.springframework:spring-core:4.2.6.RELEASE -> 4.3.14.RELEASE (*)
|    |    +--- com.fasterxml.jackson.core:jackson-databind:2.6.6 -> 2.8.10 (*)
|    |    \--- org.springframework:spring-context:4.2.6.RELEASE -> 4.3.14.RELEASE (*)
|    +--- org.springframework.integration:spring-integration-tuple:1.0.0.RELEASE
|    |    +--- org.springframework:spring-tuple:1.0.0.RELEASE (*)
|    |    \--- org.springframework.integration:spring-integration-core:4.2.5.RELEASE -> 4.3.14.RELEASE (*)
|    \--- org.springframework.retry:spring-retry:1.2.1.RELEASE -> 1.2.2.RELEASE (*)
+--- org.springframework.cloud:spring-cloud-stream-schema: -> 1.3.0.RELEASE
|    \--- org.springframework.cloud:spring-cloud-stream:1.3.0.RELEASE (*)
+--- org.springframework.cloud:spring-cloud-starter-stream-kafka: -> 1.3.0.RELEASE
|    \--- org.springframework.cloud:spring-cloud-stream-binder-kafka:1.3.0.RELEASE
|         +--- org.springframework.cloud:spring-cloud-stream-binder-kafka-core:1.3.0.RELEASE
|         |    +--- org.springframework.cloud:spring-cloud-stream:1.3.0.RELEASE (*)
|         |    +--- org.apache.kafka:kafka-clients:0.10.1.1 -> 0.11.0.1
|         |    |    +--- net.jpountz.lz4:lz4:1.3.0
|         |    |    +--- org.xerial.snappy:snappy-java:1.1.2.6
|         |    |    \--- org.slf4j:slf4j-api:1.7.25
|         |    +--- org.apache.kafka:kafka_2.11:0.10.1.1
|         |    |    +--- org.apache.kafka:kafka-clients:0.10.1.1 -> 0.11.0.1 (*)
|         |    |    +--- net.sf.jopt-simple:jopt-simple:4.9 -> 5.0.2
|         |    |    +--- com.yammer.metrics:metrics-core:2.2.0
|         |    |    |    \--- org.slf4j:slf4j-api:1.7.2 -> 1.7.25
|         |    |    +--- org.scala-lang:scala-library:2.11.8
|         |    |    +--- com.101tec:zkclient:0.9
|         |    |    |    +--- org.slf4j:slf4j-api:1.6.1 -> 1.7.25
|         |    |    |    \--- org.apache.zookeeper:zookeeper:3.4.8
|         |    |    |         +--- org.slf4j:slf4j-api:1.6.1 -> 1.7.25
|         |    |    |         \--- io.netty:netty:3.7.0.Final
|         |    |    +--- org.apache.zookeeper:zookeeper:3.4.8 (*)
|         |    |    \--- org.scala-lang.modules:scala-parser-combinators_2.11:1.0.4
|         |    |         \--- org.scala-lang:scala-library:2.11.6 -> 2.11.8
|         |    \--- org.springframework.integration:spring-integration-kafka:2.1.2.RELEASE -> 2.3.0.RELEASE
|         |         +--- org.springframework.integration:spring-integration-core:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|         |         \--- org.springframework.kafka:spring-kafka:1.3.0.RELEASE -> 1.3.4.RELEASE
|         |              +--- org.springframework:spring-messaging:4.3.14.RELEASE (*)
|         |              +--- org.springframework:spring-tx:4.3.14.RELEASE (*)
|         |              +--- org.springframework.retry:spring-retry:1.2.2.RELEASE (*)
|         |              \--- org.apache.kafka:kafka-clients:0.11.0.0 -> 0.11.0.1 (*)
|         +--- org.springframework.cloud:spring-cloud-stream:1.3.0.RELEASE (*)
|         +--- org.springframework.cloud:spring-cloud-stream-codec:1.3.0.RELEASE
|         |    +--- com.esotericsoftware:kryo-shaded:3.0.3 (*)
|         |    \--- com.fasterxml.jackson.core:jackson-annotations:2.8.0
|         +--- org.apache.kafka:kafka_2.11:0.10.1.1 (*)
|         +--- org.apache.kafka:kafka-clients:0.10.1.1 -> 0.11.0.1 (*)
|         \--- org.springframework.kafka:spring-kafka:1.1.6.RELEASE -> 1.3.4.RELEASE (*)
+--- org.springframework.cloud:spring-cloud-starter-openfeign: -> 1.4.4.RELEASE
|    +--- org.springframework.cloud:spring-cloud-starter:1.3.3.RELEASE
|    |    +--- org.springframework.boot:spring-boot-starter:1.5.10.RELEASE (*)
|    |    +--- org.springframework.cloud:spring-cloud-context:1.3.3.RELEASE
|    |    |    \--- org.springframework.security:spring-security-crypto:4.2.4.RELEASE
|    |    +--- org.springframework.cloud:spring-cloud-commons:1.3.3.RELEASE
|    |    |    +--- org.springframework.security:spring-security-crypto:4.2.4.RELEASE
|    |    |    \--- org.apache.httpcomponents:httpclient:4.5.1 -> 4.5.5
|    |    |         +--- org.apache.httpcomponents:httpcore:4.4.9
|    |    |         +--- commons-logging:commons-logging:1.2
|    |    |         \--- commons-codec:commons-codec:1.10
|    |    \--- org.springframework.security:spring-security-rsa:1.0.3.RELEASE
|    |         \--- org.bouncycastle:bcpkix-jdk15on:1.55
|    |              \--- org.bouncycastle:bcprov-jdk15on:1.55
|    +--- org.springframework.cloud:spring-cloud-netflix-core:1.4.4.RELEASE
|    |    +--- org.springframework.boot:spring-boot:1.5.10.RELEASE (*)
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:1.5.10.RELEASE (*)
|    |    \--- org.springframework.cloud:spring-cloud-commons:1.3.3.RELEASE (*)
|    +--- org.springframework:spring-web:4.3.14.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-commons:1.3.3.RELEASE (*)
|    +--- io.github.openfeign:feign-core:9.5.0
|    |    \--- org.jvnet:animal-sniffer-annotation:1.0
|    +--- io.github.openfeign:feign-slf4j:9.5.0
|    |    +--- io.github.openfeign:feign-core:9.5.0 (*)
|    |    \--- org.slf4j:slf4j-api:1.7.13 -> 1.7.25
|    +--- io.github.openfeign:feign-hystrix:9.5.0
|    |    +--- io.github.openfeign:feign-core:9.5.0 (*)
|    |    +--- com.netflix.archaius:archaius-core:0.6.6 -> 0.7.4
|    |    |    +--- com.google.code.findbugs:jsr305:3.0.1
|    |    |    +--- commons-configuration:commons-configuration:1.8
|    |    |    |    +--- commons-lang:commons-lang:2.6
|    |    |    |    \--- commons-logging:commons-logging:1.1.1 -> 1.2
|    |    |    +--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    |    |    +--- com.google.guava:guava:16.0 -> 18.0
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.4.3 -> 2.8.0
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.4.3 -> 2.8.10
|    |    |    \--- com.fasterxml.jackson.core:jackson-databind:2.4.3 -> 2.8.10 (*)
|    |    \--- com.netflix.hystrix:hystrix-core:1.4.26 -> 1.5.12
|    |         +--- org.slf4j:slf4j-api:1.7.0 -> 1.7.25
|    |         +--- com.netflix.archaius:archaius-core:0.4.1 -> 0.7.4 (*)
|    |         +--- io.reactivex:rxjava:1.2.0
|    |         \--- org.hdrhistogram:HdrHistogram:2.1.9 -> 2.1.10
|    +--- org.springframework.cloud:spring-cloud-starter-netflix-ribbon:1.4.4.RELEASE
|    |    +--- org.springframework.cloud:spring-cloud-starter:1.3.3.RELEASE (*)
|    |    +--- org.springframework.cloud:spring-cloud-netflix-core:1.4.4.RELEASE (*)
|    |    +--- org.springframework.cloud:spring-cloud-starter-netflix-archaius:1.4.4.RELEASE
|    |    |    +--- org.springframework.cloud:spring-cloud-starter:1.3.3.RELEASE (*)
|    |    |    +--- org.springframework.cloud:spring-cloud-netflix-core:1.4.4.RELEASE (*)
|    |    |    +--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    |    |    +--- commons-configuration:commons-configuration:1.8 (*)
|    |    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.8.0
|    |    |    +--- com.fasterxml.jackson.core:jackson-core:2.8.10
|    |    |    +--- com.fasterxml.jackson.core:jackson-databind:2.8.10 (*)
|    |    |    \--- com.google.guava:guava:18.0
|    |    +--- com.netflix.ribbon:ribbon:2.2.5
|    |    |    +--- com.netflix.ribbon:ribbon-core:2.2.5
|    |    |    |    +--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    |    |    |    +--- com.google.guava:guava:16.0 -> 18.0
|    |    |    |    +--- commons-configuration:commons-configuration:1.8 (*)
|    |    |    |    +--- commons-lang:commons-lang:2.6
|    |    |    |    \--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    |    |    +--- com.netflix.ribbon:ribbon-transport:2.2.5
|    |    |    |    +--- com.netflix.ribbon:ribbon-core:2.2.5 (*)
|    |    |    |    +--- com.netflix.ribbon:ribbon-loadbalancer:2.2.5
|    |    |    |    |    +--- com.netflix.ribbon:ribbon-core:2.2.5 (*)
|    |    |    |    |    +--- com.netflix.netflix-commons:netflix-statistics:0.1.1
|    |    |    |    |    |    \--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    |    |    |    |    +--- io.reactivex:rxjava:1.0.9 -> 1.2.0
|    |    |    |    |    +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
|    |    |    |    |    +--- com.netflix.servo:servo-core:0.10.1
|    |    |    |    |    |    +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
|    |    |    |    |    |    +--- com.google.guava:guava:16.0.1 -> 18.0
|    |    |    |    |    |    \--- com.netflix.servo:servo-internal:0.10.1
|    |    |    |    |    |         +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
|    |    |    |    |    |         \--- com.google.guava:guava:16.0.1 -> 18.0
|    |    |    |    |    +--- com.google.guava:guava:16.0.1 -> 18.0
|    |    |    |    |    +--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    |    |    |    |    \--- com.netflix.netflix-commons:netflix-commons-util:0.1.1
|    |    |    |    |         \--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    |    |    |    +--- io.reactivex:rxjava:1.0.10 -> 1.2.0
|    |    |    |    +--- io.reactivex:rxnetty:0.4.9
|    |    |    |    |    +--- io.reactivex:rxjava:1.0.10 -> 1.2.0
|    |    |    |    |    +--- io.netty:netty-codec-http:4.0.27.Final
|    |    |    |    |    |    +--- io.netty:netty-codec:4.0.27.Final
|    |    |    |    |    |    |    \--- io.netty:netty-transport:4.0.27.Final
|    |    |    |    |    |    |         \--- io.netty:netty-buffer:4.0.27.Final
|    |    |    |    |    |    |              \--- io.netty:netty-common:4.0.27.Final
|    |    |    |    |    |    \--- io.netty:netty-handler:4.0.27.Final
|    |    |    |    |    |         +--- io.netty:netty-buffer:4.0.27.Final (*)
|    |    |    |    |    |         +--- io.netty:netty-transport:4.0.27.Final (*)
|    |    |    |    |    |         \--- io.netty:netty-codec:4.0.27.Final (*)
|    |    |    |    |    +--- io.netty:netty-transport-native-epoll:4.0.27.Final
|    |    |    |    |    |    +--- io.netty:netty-common:4.0.27.Final
|    |    |    |    |    |    +--- io.netty:netty-buffer:4.0.27.Final (*)
|    |    |    |    |    |    \--- io.netty:netty-transport:4.0.27.Final (*)
|    |    |    |    |    \--- org.slf4j:slf4j-api:1.7.6 -> 1.7.25
|    |    |    |    +--- io.reactivex:rxnetty-contexts:0.4.9
|    |    |    |    |    +--- io.reactivex:rxjava:1.0.10 -> 1.2.0
|    |    |    |    |    \--- io.reactivex:rxnetty:0.4.9 (*)
|    |    |    |    +--- io.reactivex:rxnetty-servo:0.4.9
|    |    |    |    |    +--- io.reactivex:rxjava:1.0.10 -> 1.2.0
|    |    |    |    |    +--- io.reactivex:rxnetty:0.4.9 (*)
|    |    |    |    |    \--- com.netflix.servo:servo-core:0.7.5 -> 0.10.1 (*)
|    |    |    |    +--- javax.inject:javax.inject:1
|    |    |    |    +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
|    |    |    |    +--- com.google.guava:guava:16.0.1 -> 18.0
|    |    |    |    \--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    |    |    +--- com.netflix.hystrix:hystrix-core:1.4.3 -> 1.5.12 (*)
|    |    |    +--- javax.inject:javax.inject:1
|    |    |    +--- io.reactivex:rxjava:1.0.10 -> 1.2.0
|    |    |    +--- io.reactivex:rxnetty:0.4.9 (*)
|    |    |    +--- commons-configuration:commons-configuration:1.8 (*)
|    |    |    +--- com.google.guava:guava:16.0.1 -> 18.0
|    |    |    \--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    |    +--- com.netflix.ribbon:ribbon-core:2.2.5 (*)
|    |    +--- com.netflix.ribbon:ribbon-httpclient:2.2.5
|    |    |    +--- com.netflix.ribbon:ribbon-core:2.2.5 (*)
|    |    |    +--- com.netflix.ribbon:ribbon-loadbalancer:2.2.5 (*)
|    |    |    +--- commons-collections:commons-collections:3.2.2
|    |    |    +--- org.apache.httpcomponents:httpclient:4.2.1 -> 4.5.5 (*)
|    |    |    +--- com.sun.jersey:jersey-client:1.19.1
|    |    |    |    \--- com.sun.jersey:jersey-core:1.19.1
|    |    |    |         \--- javax.ws.rs:jsr311-api:1.1.1
|    |    |    +--- com.sun.jersey.contribs:jersey-apache-client4:1.19.1
|    |    |    |    +--- org.apache.httpcomponents:httpclient:4.1.1 -> 4.5.5 (*)
|    |    |    |    \--- com.sun.jersey:jersey-client:1.19.1 (*)
|    |    |    +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
|    |    |    +--- com.netflix.servo:servo-core:0.10.1 (*)
|    |    |    +--- com.google.guava:guava:16.0.1 -> 18.0
|    |    |    +--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    |    |    \--- com.netflix.netflix-commons:netflix-commons-util:0.1.1 (*)
|    |    +--- com.netflix.ribbon:ribbon-loadbalancer:2.2.5 (*)
|    |    \--- io.reactivex:rxjava:1.2.0
|    \--- org.springframework.cloud:spring-cloud-starter-netflix-archaius:1.4.4.RELEASE (*)
+--- org.springframework.cloud:spring-cloud-starter-netflix-eureka-client: -> 1.4.4.RELEASE
|    +--- org.springframework.boot:spring-boot-starter-web:1.5.10.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-starter:1.3.3.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-netflix-core:1.4.4.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-netflix-eureka-client:1.4.4.RELEASE
|    |    \--- org.springframework.cloud:spring-cloud-netflix-core:1.4.4.RELEASE (*)
|    +--- com.netflix.eureka:eureka-client:1.7.2
|    |    +--- org.codehaus.jettison:jettison:1.3.7
|    |    |    \--- stax:stax-api:1.0.1
|    |    +--- com.netflix.netflix-commons:netflix-eventbus:0.3.0
|    |    |    +--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    |    |    +--- com.netflix.netflix-commons:netflix-infix:0.3.0
|    |    |    |    +--- org.slf4j:slf4j-api:1.6.4 -> 1.7.25
|    |    |    |    +--- commons-jxpath:commons-jxpath:1.3
|    |    |    |    +--- joda-time:joda-time:2.3 -> 2.9.9
|    |    |    |    +--- org.antlr:antlr-runtime:3.4
|    |    |    |    |    +--- org.antlr:stringtemplate:3.2.1
|    |    |    |    |    |    \--- antlr:antlr:2.7.7
|    |    |    |    |    \--- antlr:antlr:2.7.7
|    |    |    |    +--- com.google.code.findbugs:jsr305:3.0.1
|    |    |    |    +--- com.google.guava:guava:14.0.1 -> 18.0
|    |    |    |    \--- com.google.code.gson:gson:2.1 -> 2.8.2
|    |    |    +--- com.netflix.servo:servo-core:0.5.3 -> 0.10.1 (*)
|    |    |    +--- com.netflix.archaius:archaius-core:0.3.3 -> 0.7.4 (*)
|    |    |    \--- org.apache.commons:commons-math:2.2
|    |    +--- com.thoughtworks.xstream:xstream:1.4.9 -> 1.4.10
|    |    |    +--- xmlpull:xmlpull:1.1.3.1
|    |    |    \--- xpp3:xpp3_min:1.1.4c
|    |    +--- com.netflix.archaius:archaius-core:0.7.5 -> 0.7.4 (*)
|    |    +--- javax.ws.rs:jsr311-api:1.1.1
|    |    +--- com.netflix.servo:servo-core:0.10.1 (*)
|    |    +--- com.sun.jersey:jersey-core:1.19.1 (*)
|    |    +--- com.sun.jersey:jersey-client:1.19.1 (*)
|    |    +--- com.sun.jersey.contribs:jersey-apache-client4:1.19.1 (*)
|    |    +--- org.apache.httpcomponents:httpclient:4.3.4 -> 4.5.5 (*)
|    |    +--- com.google.inject:guice:4.1.0
|    |    |    +--- javax.inject:javax.inject:1
|    |    |    +--- aopalliance:aopalliance:1.0
|    |    |    \--- com.google.guava:guava:19.0 -> 18.0
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.8.7 -> 2.8.0
|    |    +--- com.fasterxml.jackson.core:jackson-core:2.8.7 -> 2.8.10
|    |    \--- com.fasterxml.jackson.core:jackson-databind:2.8.7 -> 2.8.10 (*)
|    +--- com.netflix.eureka:eureka-core:1.7.2
|    |    +--- com.netflix.eureka:eureka-client:1.7.2 (*)
|    |    +--- com.thoughtworks.xstream:xstream:1.4.9 -> 1.4.10 (*)
|    |    +--- javax.ws.rs:jsr311-api:1.1.1
|    |    \--- org.codehaus.woodstox:woodstox-core-asl:4.4.1
|    |         +--- javax.xml.stream:stax-api:1.0-2
|    |         \--- org.codehaus.woodstox:stax2-api:3.1.4
|    +--- org.springframework.cloud:spring-cloud-starter-netflix-archaius:1.4.4.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-starter-netflix-ribbon:1.4.4.RELEASE (*)
|    +--- com.netflix.ribbon:ribbon-eureka:2.2.5
|    |    +--- com.netflix.ribbon:ribbon-core:2.2.5 (*)
|    |    +--- com.netflix.ribbon:ribbon-loadbalancer:2.2.5 (*)
|    |    +--- com.netflix.eureka:eureka-client:1.4.6 -> 1.7.2 (*)
|    |    +--- org.slf4j:slf4j-api:1.7.12 -> 1.7.25
|    |    \--- com.netflix.archaius:archaius-core:0.7.4 (*)
|    \--- com.thoughtworks.xstream:xstream:1.4.10 (*)
+--- org.springframework.cloud:spring-cloud-starter-config: -> 1.4.3.RELEASE
|    +--- org.springframework.cloud:spring-cloud-starter:1.3.3.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-config-client:1.4.3.RELEASE
|    |    +--- org.springframework.boot:spring-boot-autoconfigure:1.5.10.RELEASE (*)
|    |    +--- org.springframework.cloud:spring-cloud-commons:1.3.3.RELEASE (*)
|    |    +--- org.springframework.cloud:spring-cloud-context:1.3.3.RELEASE (*)
|    |    +--- org.springframework:spring-web:4.3.14.RELEASE (*)
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.8.0
|    |    \--- com.fasterxml.jackson.core:jackson-databind:2.8.10 (*)
|    \--- com.fasterxml.jackson.core:jackson-databind:2.8.10 (*)
+--- org.springframework.cloud:spring-cloud-starter-zipkin: -> 1.3.3.RELEASE
|    +--- org.springframework.cloud:spring-cloud-starter-sleuth:1.3.3.RELEASE
|    |    +--- org.springframework.cloud:spring-cloud-starter:1.3.3.RELEASE (*)
|    |    +--- org.springframework.boot:spring-boot-starter-aop:1.5.10.RELEASE (*)
|    |    \--- org.springframework.cloud:spring-cloud-sleuth-core:1.3.3.RELEASE
|    |         +--- org.springframework.cloud:spring-cloud-commons:1.3.3.RELEASE (*)
|    |         +--- org.springframework:spring-context:4.3.14.RELEASE (*)
|    |         \--- org.aspectj:aspectjrt:1.8.13
|    \--- org.springframework.cloud:spring-cloud-sleuth-zipkin:1.3.3.RELEASE
|         +--- org.springframework.cloud:spring-cloud-sleuth-core:1.3.3.RELEASE (*)
|         +--- org.springframework:spring-web:4.3.14.RELEASE (*)
|         +--- org.springframework.cloud:spring-cloud-commons:1.3.3.RELEASE (*)
|         +--- io.zipkin.zipkin2:zipkin:2.6.1
|         +--- io.zipkin.reporter2:zipkin-reporter:2.5.0
|         |    \--- io.zipkin.zipkin2:zipkin:2.6.1
|         +--- io.zipkin.reporter2:zipkin-sender-kafka11:2.5.0
|         |    +--- io.zipkin.reporter2:zipkin-reporter:2.5.0 (*)
|         |    \--- io.zipkin.zipkin2:zipkin:2.6.1
|         \--- io.zipkin.reporter2:zipkin-sender-amqp-client:2.5.0
|              +--- io.zipkin.reporter2:zipkin-reporter:2.5.0 (*)
|              \--- io.zipkin.zipkin2:zipkin:2.6.1
+--- org.springframework.retry:spring-retry: -> 1.2.2.RELEASE (*)
+--- org.springframework.boot:spring-boot-starter-security: -> 1.5.10.RELEASE
|    +--- org.springframework.boot:spring-boot-starter:1.5.10.RELEASE (*)
|    +--- org.springframework:spring-aop:4.3.14.RELEASE (*)
|    +--- org.springframework.security:spring-security-config:4.2.4.RELEASE
|    |    +--- org.springframework.security:spring-security-core:4.2.4.RELEASE
|    |    |    +--- org.springframework:spring-aop:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    |    +--- org.springframework:spring-beans:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    |    +--- org.springframework:spring-context:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    |    +--- org.springframework:spring-core:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    |    \--- org.springframework:spring-expression:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-aop:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-beans:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    +--- org.springframework:spring-context:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    |    \--- org.springframework:spring-core:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|    \--- org.springframework.security:spring-security-web:4.2.4.RELEASE
|         +--- org.springframework.security:spring-security-core:4.2.4.RELEASE (*)
|         +--- org.springframework:spring-beans:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|         +--- org.springframework:spring-context:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|         +--- org.springframework:spring-core:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|         +--- org.springframework:spring-expression:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
|         \--- org.springframework:spring-web:4.3.12.RELEASE -> 4.3.14.RELEASE (*)
+--- org.springframework.kafka:spring-kafka:1.3.4.RELEASE (*)
+--- org.springframework.integration:spring-integration-kafka:2.3.0.RELEASE (*)
+--- org.apache.kafka:kafka-clients:0.11.0.1 (*)
+--- io.springfox:springfox-swagger2:2.7.0
|    +--- io.swagger:swagger-annotations:1.5.13
|    +--- io.swagger:swagger-models:1.5.13
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.8.5 -> 2.8.0
|    |    +--- org.slf4j:slf4j-api:1.7.22 -> 1.7.25
|    |    \--- io.swagger:swagger-annotations:1.5.13
|    +--- io.springfox:springfox-spi:2.7.0
|    |    \--- io.springfox:springfox-core:2.7.0
|    |         +--- net.bytebuddy:byte-buddy:1.6.14
|    |         +--- com.google.guava:guava:18.0
|    |         +--- com.fasterxml:classmate:1.3.3 -> 1.3.4
|    |         +--- org.slf4j:slf4j-api:1.7.24 -> 1.7.25
|    |         +--- org.springframework.plugin:spring-plugin-core:1.2.0.RELEASE
|    |         |    +--- org.springframework:spring-beans:4.0.9.RELEASE -> 4.3.14.RELEASE (*)
|    |         |    +--- org.springframework:spring-context:4.0.9.RELEASE -> 4.3.14.RELEASE (*)
|    |         |    +--- org.springframework:spring-aop:4.0.9.RELEASE -> 4.3.14.RELEASE (*)
|    |         |    \--- org.slf4j:slf4j-api:1.7.10 -> 1.7.25
|    |         \--- org.springframework.plugin:spring-plugin-metadata:1.2.0.RELEASE
|    |              +--- org.springframework.plugin:spring-plugin-core:1.2.0.RELEASE (*)
|    |              \--- org.slf4j:slf4j-api:1.7.10 -> 1.7.25
|    +--- io.springfox:springfox-schema:2.7.0
|    |    +--- io.springfox:springfox-core:2.7.0 (*)
|    |    \--- io.springfox:springfox-spi:2.7.0 (*)
|    +--- io.springfox:springfox-swagger-common:2.7.0
|    |    +--- io.swagger:swagger-annotations:1.5.13
|    |    +--- io.swagger:swagger-models:1.5.13 (*)
|    |    +--- io.springfox:springfox-spi:2.7.0 (*)
|    |    +--- io.springfox:springfox-schema:2.7.0 (*)
|    |    +--- io.springfox:springfox-spring-web:2.7.0
|    |    |    +--- org.reflections:reflections:0.9.11
|    |    |    |    +--- com.google.guava:guava:20.0 -> 18.0
|    |    |    |    \--- org.javassist:javassist:3.21.0-GA
|    |    |    +--- com.google.guava:guava:18.0
|    |    |    +--- com.fasterxml:classmate:1.3.3 -> 1.3.4
|    |    |    +--- org.slf4j:slf4j-api:1.7.24 -> 1.7.25
|    |    |    +--- org.springframework.plugin:spring-plugin-core:1.2.0.RELEASE (*)
|    |    |    +--- org.springframework.plugin:spring-plugin-metadata:1.2.0.RELEASE (*)
|    |    |    \--- io.springfox:springfox-spi:2.7.0 (*)
|    |    +--- com.google.guava:guava:18.0
|    |    +--- com.fasterxml:classmate:1.3.3 -> 1.3.4
|    |    +--- org.slf4j:slf4j-api:1.7.24 -> 1.7.25
|    |    +--- org.springframework.plugin:spring-plugin-core:1.2.0.RELEASE (*)
|    |    \--- org.springframework.plugin:spring-plugin-metadata:1.2.0.RELEASE (*)
|    +--- io.springfox:springfox-spring-web:2.7.0 (*)
|    +--- com.google.guava:guava:18.0
|    +--- com.fasterxml:classmate:1.3.3 -> 1.3.4
|    +--- org.slf4j:slf4j-api:1.7.24 -> 1.7.25
|    +--- org.springframework.plugin:spring-plugin-core:1.2.0.RELEASE (*)
|    +--- org.springframework.plugin:spring-plugin-metadata:1.2.0.RELEASE (*)
|    \--- org.mapstruct:mapstruct:1.1.0.Final
+--- io.springfox:springfox-swagger-ui:2.7.0
|    \--- io.springfox:springfox-spring-web:2.7.0 (*)
+--- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.9.3
|    +--- com.fasterxml.jackson.core:jackson-annotations:2.9.0 -> 2.8.0
|    +--- com.fasterxml.jackson.core:jackson-core:2.9.3 -> 2.8.10
|    \--- com.fasterxml.jackson.core:jackson-databind:2.9.3 -> 2.8.10 (*)
+--- com.fasterxml.jackson.dataformat:jackson-dataformat-avro:2.8.10
|    +--- com.fasterxml.jackson.core:jackson-databind:2.8.10 (*)
|    +--- org.apache.avro:avro:1.7.7 -> 1.8.2
|    |    +--- org.codehaus.jackson:jackson-core-asl:1.9.13
|    |    +--- org.codehaus.jackson:jackson-mapper-asl:1.9.13
|    |    |    \--- org.codehaus.jackson:jackson-core-asl:1.9.13
|    |    +--- com.thoughtworks.paranamer:paranamer:2.7
|    |    +--- org.xerial.snappy:snappy-java:1.1.1.3 -> 1.1.2.6
|    |    +--- org.apache.commons:commons-compress:1.8.1
|    |    +--- org.tukaani:xz:1.5
|    |    \--- org.slf4j:slf4j-api:1.7.7 -> 1.7.25
|    \--- com.fasterxml.jackson.core:jackson-core:2.8.10
+--- org.apache.avro:avro:1.8.2 (*)
+--- org.elasticsearch:elasticsearch:6.2.2
|    +--- org.elasticsearch:elasticsearch-core:6.2.2
|    |    \--- org.apache.logging.log4j:log4j-api:2.9.1 -> 2.7
|    +--- org.apache.lucene:lucene-core:7.2.1
|    +--- org.apache.lucene:lucene-analyzers-common:7.2.1
|    +--- org.apache.lucene:lucene-backward-codecs:7.2.1
|    +--- org.apache.lucene:lucene-grouping:7.2.1
|    +--- org.apache.lucene:lucene-highlighter:7.2.1
|    +--- org.apache.lucene:lucene-join:7.2.1
|    +--- org.apache.lucene:lucene-memory:7.2.1
|    +--- org.apache.lucene:lucene-misc:7.2.1
|    +--- org.apache.lucene:lucene-queries:7.2.1
|    +--- org.apache.lucene:lucene-queryparser:7.2.1
|    +--- org.apache.lucene:lucene-sandbox:7.2.1
|    +--- org.apache.lucene:lucene-spatial:7.2.1
|    +--- org.apache.lucene:lucene-spatial-extras:7.2.1
|    +--- org.apache.lucene:lucene-spatial3d:7.2.1
|    +--- org.apache.lucene:lucene-suggest:7.2.1
|    +--- org.elasticsearch:securesm:1.2
|    +--- org.elasticsearch:elasticsearch-cli:6.2.2
|    |    \--- net.sf.jopt-simple:jopt-simple:5.0.2
|    +--- com.carrotsearch:hppc:0.7.1
|    +--- joda-time:joda-time:2.9.9
|    +--- org.yaml:snakeyaml:1.17
|    +--- com.fasterxml.jackson.core:jackson-core:2.8.10
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-smile:2.8.10
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.8.10
|    +--- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.8.10
|    +--- com.tdunning:t-digest:3.0
|    +--- org.hdrhistogram:HdrHistogram:2.1.9 -> 2.1.10
|    +--- org.apache.logging.log4j:log4j-api:2.9.1 -> 2.7
|    \--- org.elasticsearch:jna:4.5.1
+--- org.elasticsearch.client:elasticsearch-rest-high-level-client:6.2.2
|    +--- org.elasticsearch:elasticsearch:6.2.2 (*)
|    +--- org.elasticsearch.client:elasticsearch-rest-client:6.2.2
|    |    +--- org.apache.httpcomponents:httpclient:4.5.2 -> 4.5.5 (*)
|    |    +--- org.apache.httpcomponents:httpcore:4.4.5 -> 4.4.9
|    |    +--- org.apache.httpcomponents:httpasyncclient:4.1.2 -> 4.1.3
|    |    +--- org.apache.httpcomponents:httpcore-nio:4.4.5
|    |    +--- commons-codec:commons-codec:1.10
|    |    \--- commons-logging:commons-logging:1.1.3 -> 1.2
|    +--- org.elasticsearch.plugin:parent-join-client:6.2.2
|    |    +--- org.elasticsearch:elasticsearch:6.2.2 (*)
|    |    +--- org.locationtech.spatial4j:spatial4j:0.6
|    |    +--- com.vividsolutions:jts:1.13
|    |    +--- org.apache.logging.log4j:log4j-api:2.9.1 -> 2.7
|    |    +--- org.apache.logging.log4j:log4j-core:2.9.1 -> 2.7
|    |    \--- org.elasticsearch:jna:4.5.1
|    +--- org.elasticsearch.plugin:aggs-matrix-stats-client:6.2.2
|    |    +--- org.elasticsearch:elasticsearch:6.2.2 (*)
|    |    +--- org.locationtech.spatial4j:spatial4j:0.6
|    |    +--- com.vividsolutions:jts:1.13
|    |    +--- org.apache.logging.log4j:log4j-api:2.9.1 -> 2.7
|    |    +--- org.apache.logging.log4j:log4j-core:2.9.1 -> 2.7
|    |    \--- org.elasticsearch:jna:4.5.1
|    \--- org.elasticsearch.plugin:rank-eval-client:6.2.2
|         +--- org.elasticsearch:elasticsearch:6.2.2 (*)
|         +--- org.locationtech.spatial4j:spatial4j:0.6
|         +--- com.vividsolutions:jts:1.13
|         +--- org.apache.logging.log4j:log4j-api:2.9.1 -> 2.7
|         +--- org.apache.logging.log4j:log4j-core:2.9.1 -> 2.7
|         \--- org.elasticsearch:jna:4.5.1
+--- <our Java library>
|    +--- org.springframework:spring-web:4.3.14.RELEASE (*)
|    +--- org.springframework:spring-context:4.3.14.RELEASE (*)
|    \--- org.hibernate:hibernate-validator:5.3.6.Final (*)
+--- <our Java library>
|    +--- org.springframework.boot:spring-boot-starter-web:1.5.10.RELEASE (*)
|    \--- org.projectlombok:lombok:1.16.20
+--- <our Java library>
|    +--- org.springframework.boot:spring-boot-starter-actuator:1.5.10.RELEASE (*)
|    +--- org.springframework.cloud:spring-cloud-commons:1.3.2.RELEASE -> 1.3.3.RELEASE (*)
|    +--- org.springframework:spring-web:4.3.14.RELEASE (*)
|    +--- io.micrometer:micrometer-spring-legacy:1.0.0
|    |    \--- io.micrometer:micrometer-core:1.0.0
|    |         +--- org.hdrhistogram:HdrHistogram:2.1.10
|    |         \--- org.latencyutils:LatencyUtils:2.0.3
|    +--- io.micrometer:micrometer-registry-prometheus:1.0.0
|    |    +--- io.micrometer:micrometer-core:1.0.0 (*)
|    |    \--- io.prometheus:simpleclient_common:0.2.0
|    |         \--- io.prometheus:simpleclient:0.2.0
|    \--- org.projectlombok:lombok:1.16.20
+--- org.apache.commons:commons-lang3:3.7
\--- net.logstash.logback:logstash-logback-encoder:4.9
     \--- com.fasterxml.jackson.core:jackson-databind:2.6.5 -> 2.8.10 (*)
```

**Comment by garyrussell** [comment]
>`|         +--- org.apache.kafka:kafka_2.11:0.10.1.1 (*)`

Try 0.11.0.1 for the scala jar, or drop the kafka-clients back to 0.10.1.1.

**Comment by sabbyanandan** [comment]
@lubos-bistak: If you have any other questions, please open a new issue or at StackOverflow.
